compose/src/main/java/com/bottlerocketstudios/compose/snippets/snippetDetails/snippetDetailsWidgets/SnippetDetailsFileCard.kt

```diff
@@ -0,0 +1,123 @@
+import com.bottlerocketstudios.compose.util.convertToImageBitmap
+import com.bottlerocketstudios.compose.widgets.IconText
+import com.bottlerocketstudios.launchpad.compose.light
+import com.bottlerocketstudios.launchpad.compose.normal
+
+@Composable
+fun SnippetDetailsFilesCard(file: SnippetDetailsFile?) {
+    var expanded by remember { mutableStateOf(false) }
+
+    Card(
+        elevation = Dimens.plane_3,
+        modifier = Modifier
+            .fillMaxWidth()
+            .wrapContentHeight()
+            .padding(vertical = Dimens.grid_1, horizontal = Dimens.grid_2)
+    ) {
+        Column(
+            modifier = Modifier.animateContentSize(tween(1000))
+        ) {
+            Row(
+                modifier = Modifier
+                    .padding(
+                        vertical = Dimens.grid_1,
+                        horizontal = Dimens.grid_2
+                    )
+                    .fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                IconText(
+                    iconRes = R.drawable.ic_file,
+                    iconColor = Colors.tertiary,
+                    text = file?.fileName ?: "",
+                    style = typography.h4.normal()
+                )
+                OutlinedButton(
+                    onClick = { expanded = !expanded },
+                    colors = ButtonDefaults.buttonColors(backgroundColor = Colors.surface),
+                    elevation = null
+                ) {
+                    Text(
+                        text = if (expanded) "Close" else "RAW",
+                        style = typography.body1.copy(color = Colors.onSurface)
+                    )
+                }
+            }
+
+            if (expanded) {
+                Divider(
+                    color = brown_grey_three,
+                    modifier = Modifier
+                        .padding(horizontal = Dimens.grid_2)
+                        .wrapContentHeight()
+                        .fillMaxWidth()
+                )
+                RawFileData(byteArray = file?.rawFile)
+            }
+        }
+    }
+}
+
+@Composable
+fun RawFileData(byteArray: ByteArray?) {
+    byteArray?.let {
+        val imageBitmap = byteArray.convertToImageBitmap()
+        if (imageBitmap != null) {
+            Image(
+                imageBitmap,
+                contentDescription = stringResource(id = R.string.file_image_description),
+                modifier = Modifier.padding(Dimens.grid_2)
+            )
+        } else {
+            Text(
+                text = byteArray.decodeToString(),
+                style = MaterialTheme.typography.h5.light(),
+                modifier = Modifier
+                    .padding(Dimens.grid_2)
+                    .fillMaxWidth()
+            )
+        }
+    }
+}
+
+@Preview(showSystemUi = true)
+@Composable
+fun PreviewFilesCard() {
```

Comment:
@MMurrayAtBR Make `private`

```suggestion
private fun PreviewFilesCard() {
```